tasks/actorstate/verifreg.go

```diff
@@ -0,0 +1,240 @@
+				Address:   addr.String(),
+			})
+			return nil
+		}); err != nil {
+			return nil, nil, err
+		}
+		return verifiers, events, nil
+	}
+
+	changes, err := verifreg.DiffVerifiers(ctx, ec.Store, ec.PrevState, ec.CurrState)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("diffing verifier registry verifiers: %w", err)
+	}
+
+	// a new verifier was added
+	for _, change := range changes.Added {
+		verifiers = append(verifiers, &verifregmodel.VerifiedRegistryVerifier{
+			Height:    int64(ec.CurrTs.Height()),
+			StateRoot: ec.CurrTs.ParentState().String(),
+			Address:   change.Address.String(),
+			DataCap:   change.DataCap.String(),
+		})
+		events = append(events, &verifregmodel.VerifiedRegistryVerifierEvent{
+			Height:    int64(ec.CurrTs.Height()),
+			StateRoot: ec.CurrTs.ParentState().String(),
+			Event:     verifregmodel.Added,
+			Address:   change.Address.String(),
+		})
+	}
+	// a verifier was removed
+	for _, change := range changes.Removed {
+		events = append(events, &verifregmodel.VerifiedRegistryVerifierEvent{
+			Height:    int64(ec.CurrTs.Height()),
+			StateRoot: ec.CurrTs.ParentState().String(),
+			Event:     verifregmodel.Removed,
+			Address:   change.Address.String(),
+		})
+	}
+	// an existing verifier's DataCap changed
+	for _, change := range changes.Modified {
+		verifiers = append(verifiers, &verifregmodel.VerifiedRegistryVerifier{
+			Height:    int64(ec.CurrTs.Height()),
+			StateRoot: ec.CurrTs.ParentState().String(),
+			Address:   change.After.Address.String(),
+			DataCap:   change.After.DataCap.String(),
+		})
+	}
+	return verifiers, events, nil
+}
+
+func ExtractVerifiedClients(ctx context.Context, ec *VerifiedRegistryExtractionContext) (verifregmodel.VerifiedRegistryVerifiedClientsList, verifregmodel.VerifiedRegistryClientEventList, error) {
+	var clients verifregmodel.VerifiedRegistryVerifiedClientsList
+	var events verifregmodel.VerifiedRegistryClientEventList
+	// if this is the genesis state extract whatever state it has, there is noting to diff against
+	if !ec.HasPreviousState() {
+		if err := ec.CurrState.ForEachClient(func(addr address.Address, dcap abi.StoragePower) error {
+			clients = append(clients, &verifregmodel.VerifiedRegistryVerifiedClient{
+				Height:    int64(ec.CurrTs.Height()),
+				StateRoot: ec.CurrTs.ParentState().String(),
+				Address:   addr.String(),
+				DataCap:   dcap.String(),
+			})
+			events = append(events, &verifregmodel.VerifiedRegistryClientEvent{
+				Height:    int64(ec.CurrTs.Height()),
+				StateRoot: ec.CurrTs.ParentState().String(),
+				Event:     verifregmodel.Added,
+				Address:   addr.String(),
+			})
+			return nil
+		}); err != nil {
+			return nil, nil, err
+		}
+		return clients, events, nil
+	}
+
+	changes, err := verifreg.DiffVerifiedClients(ctx, ec.Store, ec.PrevState, ec.CurrState)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("diffing verifier registry clients: %w", err)
+	}
+
+	for _, change := range changes.Added {
```

Comment:
Same comment about merging the Event model into the VerifiedRegistryVerifiedClient model.